### PR TITLE
Fixes for new clippy lints in Rust 1.54.

### PR DESCRIPTION
**Pull request by cessen**

### CODE REVIEW

**Review by CBenoit — APPROVED** [review]
LGTM

File: helix-core/src/movement.rs [hunk 1]
```diff
@@ -245,6 +245,7 @@ impl CharHelpers for Chars<'_> {
 
         // Find our target position(s).
         let head_start = head;
+        #[allow(clippy::while_let_on_iterator)] // Clippy's suggestion to fix doesn't work here.
```
Comment (pickfire) [inline helix-core/src/movement.rs:248]: I wonder if this should be reported to clippy as false positive?

Comment (cessen) [reply helix-core/src/movement.rs:248]: I was planning to do that, but hadn't gotten around to it yet.  Thanks for finding that issue and adding our situation to the list!

Comment (pickfire) [reply helix-core/src/movement.rs:248]: This should be fixed on next clippy release.

**Pull request merged**